mantidproject/mantid
# EnginXCalibrate: add option to produce table with output parameters

This fixes #11806  Adds a new optional property. If you give a name, a table workspace will be created (with that name) that contains the (for now) two output parameters from EnginXCalibrate.

**To test**:
- Tests should pass. The new option is tested in a doc-test but I refrained from adding a test case for it in the unit test as that would further slow it down
- Code review.
- See if you like the option added. There would be other alternatives but this one looks simple and clean.
- You can also double-check if it works locally with commands like:

```
difc, zero = EnginXCalibrate(Filename="ENGINX00213855.nxs", ExpectedPeaks=[1.097, 2.1], Bank=1, OutputParametersTableName='foo_params_table') 
difc, zero = EnginXCalibrate(Filename="ENGINX00213855.nxs", ExpectedPeaks=[1.097, 2.1], Bank=1)  # no table produced
```

**Relase notes**: at the moment I'm not adding anything about these EnginX changes, as this is very much work in progress. I'd wait until we have a cleared view on what will be the more final directions and outcomes, etc. Then we can write more sensible release notes about EnginX.
